OneSignalSDK/onesignal/src/main/java/com/onesignal/OSSystemConditionController.java

```diff
@@ -18,13 +26,51 @@
 
     boolean systemConditionsAvailable() {
         if (ActivityLifecycleHandler.curActivity == null) {
+            OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "OSSystemConditionObserver curActivity null");
             return false;
         }
+
+        if (isDialogFragmentVisible(ActivityLifecycleHandler.curActivity)) {
+            OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "OSSystemConditionObserver dialog fragment detected");
+            return false;
+        }
+
         boolean keyboardUp = OSViewUtils.isKeyboardUp(new WeakReference<>(ActivityLifecycleHandler.curActivity));
         if (keyboardUp) {
             ActivityLifecycleHandler.setSystemConditionObserver(TAG, systemConditionObserver);
+            OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "OSSystemConditionObserver keyboard up detected");
```

Comment:
Change to WARNING and explain that IAMs will not be shown

Reply:
I will prefer to have this warning independent from IAM, because this can be potentially be used in another place, no necessary only on IAM. I will add a log on the IAMController regarding this